### ISSUE TITLE
TbSync 4.2 doesn't install on TB 102.3.3

### ISSUE THREAD

**Issue by WRatzka** [jobisoft/TbSync#616]
From naively reading the manifest/application.ini:
Could this be due to the fact that TbSync asks for Gecko 102.3.3 whereas Thunderbird 102.3.3 only provides Gecko 102.3.0?

Kind regards
Wolfgang

## Your environment

TbSync version: 4.2
Thunderbird version: 102.3.3

## Expected behavior
Addon can be installed

## Actual behavior
Error message "not compatible with Thunderbird 102.3.3"

## Steps to reproduce
download TbSync.xpi, use install-addon-from-file dialog

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by jobisoft** [comment]
Looking into that right now.

**Comment by izaxone** [comment]
I am also seeing this behavior when using version 102.3.1

**Comment by cmeesters** [comment]
same for me with v102.3.0 (Debian, Bullseye - so update from package always awaiting a maintenance/maintainer cycle)

**Comment by jobisoft** [comment]
v4.3 has been released (and send to ATN for review). Should work now.

**Comment by jobisoft** [comment]
> same for me with v102.3.0 (Debian, Bullseye - so update from package always awaiting a maintenance/maintainer cycle)
> 
> PS The manifest.json states: `"gecko": { "id": "tbsync@jobisoft.de", "strict_min_version": "102.3.3", "strict_max_version": "102.*" }`

It will not work for you. The version really is for 102.3.3 or newer. What went wrong here is that the add-on system cannot use the third level in the version specifier to specify subversions, so even if I requested 102.3.3 and running 102.3.3, the add-on install check only sees 102.3.0 as the app version and fails to install the add-on.

In v4.3 I reduced the requested version to 102.3.0, but now included a manual check if the real version is 102.3.3. If not, you will get a popup, asking you to update to 102.3.3 (and the add-on will not register the provider).
